packages/next/src/client/components/navigation.ts

```diff
@@ -167,11 +169,25 @@ function getSelectedParams(
 export function useParams(): Params {
   clientHookInServerComponentError('useParams')
   const globalLayoutRouterContext = useContext(GlobalLayoutRouterContext)
-  if (!globalLayoutRouterContext) {
-    // This only happens in `pages`. Type is overwritten in navigation.d.ts
-    return null!
+  const pagesRouter = useContext(RouterContext)
```

Comment:
This should not be needed if you provide the path through context in `pages` instead, as this increases bundle size for `app` to support `pages`.

Reply:
we also need to access `router.query` from pages router, is there any other way to work around the `query`?